components/button-group/src/ButtonGroup.js

```diff
@@ -37,6 +37,18 @@ export const ButtonGroup = ({
 		setChecked(event.target.value);
 	};
 
+	const numChildren = Children.count(children);
+	const childrenWithProps = Children.map(children, (child, index) => {
+		if (index === 0) {
+			return cloneElement(child, { first: true })
+		}
+		if (index === numChildren - 1) {
+			return cloneElement(child, { last: true })
+		}
+
+		return child;
+	});
```

Comment:
As per our discussion + my overnight epiphany - this is actually a good use case for `cloneElement`.

Also, multi-line comments FTW!

Reply:
🤣